test/billing-agreements.test.js

```diff
@@ -0,0 +1,74 @@
+import { expect } from 'chai';
+import { describe, it } from 'mocha';
+import { run, generateMeta } from '@syncano/test';
+import url from 'url';
+import 'dotenv/config';
+
+import { create_billing_agreement_details, update_billing_plan_details } from './utils/helpers';
+
+describe('billing-agreements', () => {
+  const meta = generateMeta('billing_agreements');
+
+  describe('POST', () => {
+    it('should create billing agreement successfully with valid parameters', (done) => {
+      meta.request.REQUEST_METHOD = 'POST';
+      create_billing_agreement_details.plan.id = process.env.TEST_BILLING_PLAN_ID;
+      run('billing_agreements',
+        { args: { create_billing_agreement_details }, meta })
+        .then((res) => {
+          const billingAgreement = res.data;
+
+          for (let index = 0; index < billingAgreement.links.length; index += 1) {
```

Comment:
Use something nicer for finding the approval_url: array `find()` method or something like this pattern:
`arr.filter(obj => Object.keys(obj).some(key => obj[key].includes(searchKey)));`